crates/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -18,16 +28,33 @@ use self::ast::{Definition, FuncId, Function, LocalId, Program};
 pub mod ast;
 pub mod printer;
 
+/// The context struct for the monomorphization pass.
+///
+/// This uses the NodeInterner internally to reference existing definitions in the HIR,
+/// but since the produced monomorphized AST uses fresh IDs and is self-contained, there
+/// is no more need for the NodeInterner after this pass.
```

Comment:
I think we can move the first half of this to the `nodeInterner` field and remove the rest. We document that we don't use the `NodeInterner` after monomorphisation in a few places.